sdk/src/main/java/com/mapbox/maps/renderer/MapboxRenderThread.kt

```diff
@@ -219,43 +260,46 @@ internal class MapboxRenderThread : Choreographer.FrameCallback {
   }
 
   private fun releaseEglSurface() {
-    eglSurface?.let {
-      eglCore.releaseSurface(it)
-    }
-    eglSurface = null
+    eglCore.releaseSurface(eglSurface)
+    eglContextCreated = false
+    eglSurface = eglCore.eglNoSurface
   }
 
   private fun releaseAll() {
-    mapboxRenderer.onSurfaceDestroyed()
-    nativeRenderCreated = false
+    renderDestroyCallChain = true
+    mapboxRenderer.destroyRenderer()
+    renderDestroyCallChain = false
+    renderCreated = false
     releaseEgl()
     surface?.release()
   }
 
   private fun prepareRenderFrame(creatingSurface: Boolean = false) {
+    // no need to do anything if we're waiting for next VSYNC already
+    if (awaitingNextVsync) {
+      return
+    }
     // Check first if we have to stop rendering at all (even if there was no EGL config) and cleanup EGL.
     // We need to check it ASAP in order not to block thread that is calling `onSurfaceTextureDestroyed`.
     // After that check MapView could be actually rendered on this device (has valid EGL config).
     // After that we check if activity / fragment is paused.
-    if (shouldExit || nativeRenderNotSupported || paused) {
-      if (paused) {
-        needRenderOnResume = true
-      }
+    if (renderNotSupported || paused) {
       // at least on Android 8 devices we create surface before Activity#onStart
       // so we need to proceed to EGL creation in any case to avoid deadlock
       if (!creatingSurface) {
         return
       }
     }
-    if (!checkSurfaceReady(creatingSurface)) {
-      return
+    // check for creatingSurface flag to make sure we don't hit deadlock
+    if (creatingSurface || !renderThreadPrepared) {
+      val renderThreadPreparedOk = setUpRenderThread(creatingSurface)
+      if (!renderThreadPreparedOk) {
+        return
+      }
     }
     checkSurfaceSizeChanged()
-    drainQueue(renderEventQueue)
-    // listen to next VSYNC event if not listening already
-    if (awaitingNextVsync.compareAndSet(false, true)) {
-      Choreographer.getInstance().postFrameCallback(this)
-    }
+    Choreographer.getInstance().postFrameCallback(this)
```

Comment:
Is it still possible awaitingNextVsync will still be false but frame callback will already fire?